recipes-ni/ni-base-system-image-tests/files/fs_permissions_known.py

```diff
@@ -0,0 +1,162 @@
+import argparse
+import os
+import sys
+import hashlib
+import stat
+import pwd
+import grp
+import fs_permissions_shared
+
+# Function to return tree instead of variable since it contains functions inside its structure
+def known_permissions_tree():
+    return {
+        '.': system_dir,
+        'boot': {
+            '.': system_dir,
+            'bootmode': system_file,
+            'grub': {
+                '.': system_dir,
+                '*': system_file,
+                'grubenv': permissions(0o664, 'admin', 'ni', FT_REG),
+                'grubenv.bak': permissions(0o664, 'admin', 'ni', FT_REG),
+                'recoverytool-ni-version': permissions(0o0444, 'admin', 'administrators', FT_REG)
+            }, 'runmode': {
+                '.': system_dir,
+                '*': system_file,
+                'bzImage': system_link(f'bzImage-{bash("$(uname -r)")[0]}'),
+            }
+        }, 'lib': {
+            '.': system_dir,
+            'modules': {
+                '.': system_dir,
+                '$(uname -r)': {
```

Comment:
May have intended this:
```suggestion
                bash('$(uname -r)')[0]: {
```

Reply:
No because paths are [already run through bash](https://github.com/ni/meta-nilrt/pull/557/files#diff-0f9a20fa2e3b0a91cca446d50f2a12047736d56a3439f8bf4df2eb508d30d528R53).

When we traverse the file tree here, it has '.', and that '.' becomes '$(uname -r)' and is then passed through bash.

Reply:
In that case, you're missing the run-through-bash step a couple lines prior to where you linked, for the `dict` case.

Reply:
Agh, never mind--it'll get expanded per-file. Perhaps a bit wasteful, but not incorrect.